RecoPixelVertexing/PixelTriplets/plugins/GPUSimpleVector.h

```diff
@@ -0,0 +1,55 @@
+//
+// Author: Felice Pantaleo, CERN
+//
+
+#ifndef GPU_SIMPLEVECTOR_H_
+#define GPU_SIMPLEVECTOR_H_
+
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+template <int maxSize, class T> struct GPUSimpleVector {
```

Comment:
How is this different from
https://github.com/cms-patatrack/cmssw/blob/CMSSW_10_2_X_Patatrack/HeterogeneousCore/CUDAUtilities/interface/GPUSimpleVector.h
? (ok, I see `int maxSize` template parameter)

Anyway it would be better placed (eventually) in `HeterogeneousCore/CUDAUtilities`.

Reply:
Indeed, can we avoid having two `GPUSimpleVector` classes ?

Reply:
We discussed with Felice and I agree the use case is valid (one GPU allocation vs. a GPU allocation per hit). The intention of the class is to provide a vector-like interface on top of an (dynamically-allocated) array, close to what
https://github.com/cms-patatrack/cmssw/blob/655e4ed3e117cb7dfdec85dc9ebe7baf2d3ce630/FWCore/Utilities/interface/VecArray.h#L27-L28
does in CPU.

I'd suggest to treat this class similarly, i.e. rename to e.g. `GPUVecArray` (and reorder the template parameters as `<class, int>`) and move to `HeterogeneousCore/CUDAUtilities/interface`.